docs/source/storage.rst

```diff
@@ -1,5 +1,122 @@
 torch.Storage
-===================================
+=============
+
+In PyTorch, a regular tensor is a multi-dimensional array that is defined by the following components:
+
+  - Storage: The actual data of the tensor, stored as a contiguous, one-dimensional array of bytes.
+  - ``dtype``: The data type of the elements in the tensor, such as torch.float32 or torch.int64.
+  - ``shape``: A tuple indicating the size of the tensor in each dimension.
+  - Stride: The step size needed to move from one element to the next in each dimension.
+  - Offset: The starting point in the storage from which the tensor data begins.
+
+These components together define the structure and data of a tensor, with the storage holding the
+actual data and the rest serving as metadata.
+
+Untyped Storage API
+-------------------
+
+An untyped storage is a fundamental concept in PyTorch, representing a contiguous, one-dimensional array of bytes.
+It serves as the underlying data container for tensors.
+A :class:`torch.UntypedStorage` is a contiguous, one-dimensional array of elements. Its length is equal to the number of
+bytes of the tensor.
+In general, a tensor created in PyTorch using regular constructors such as :func:`~torch.zeros`, :func:`~torch.zeros_like`
+or :func:`~torch.Tensor.new_zeros` will produce tensors where there is a one-to-one correspondence between the tensor
+storage and the tensor itself.
+
+But a storage also allows multiple tensors to share the same storage.
+For instance, any view of a tensor (obtained through :meth:`~torch.Tensor.view` or some but not all kinds of indexing
+like integers and slices) will point to the same underlying tensor as the original one.
+When serializing and deserializing tensors that share a common storage, the relationship is preserved, and the tensors
+continue to point to the same storage. Interestingly, deserializing multiple tensors that point to a single storage
+can be faster than deserializing multiple independent tensors.
+
+A tensor storage can be accessed through the :meth:`~torch.Tensor.untyped_storage` method. This will return an object of
+type :class:`torch.UntypedStorage`.
+Fortunately, storages have a unique identifier called accessed through the :meth:`torch.UntypedStorage.data_ptr` method.
+In regular settings, two tensors with the same data storage will have the same storage ``data_ptr``. However, this is
+not true for the tensors themselves: for instance, a tensor with a ``data`` and a ``grad`` value actually points to two
+independent storages and hence require a ``data_ptr()`` of its own.
```

Comment:
This sentence is unclear. Maybe  " However, tensors themselves can point to two separate storages, one for its ``data`` attribute and another for its ``grad`` attribute. Each will require a ``data_ptr()`` of its own.